draft-law-moq-cat4moqt.md

```diff
@@ -106,56 +106,72 @@ The "moqt" claim is defined by the following CDDL:
 
 ~~~~~~~~~~~~~~~
 $$Claims-Set-Claims //= (moqt-label => moqt-value)
-moqt-label = XXX TODO - how do we register this?
-moqt-value = [ + moqt-object ]
+moqt-label = TBD_MOQT
+moqt-value = [ + moqt-limit ]
+moqt-limit = [ moqt-actions, moqt-ns-match, moqt-track-match ]
+moqt-actions = moqt-action | [ + moqt-action ]
+moqt-action = int
+moqt-ns-match = match
+moqt-track-match = match
+
+/ match defined in CTA-5007-B 4.6.1, repeated here for convenience /
+match = {
+  ? exact-match ^ => tstr,
+  ? prefix-match ^ => tstr,
+  ? suffix-match ^ => tstr,
+  ? contains-match ^ => tstr,
+  ? regex-match ^ => [ tstr, * tstr ],
+  ? sha256-match ^ => bstr,
+  ? sha512-256-match ^ => bstr
+}
+
+exact-match = 0
+prefix-match = 1
+suffix-match = 2
+contains-match = 3
+regex-match = 4
+sha256-match = -1
+sha512-256-match = -2
 ~~~~~~~~~~~~~~~
 
-TODO - need CDDL valid definition. The moqt token needs to encode multiple instances of 4 actions, currently
-
-* 0 - ANNOUNCE
-* 1 - SUBSCRIBE_ANNOUNCES
-* 2 - PUBLISH
-* 3 - FETCH
-
-For each action, we need to communicate the permission
-
-* 0 - Allowed for all Namespaces and Names
-* 1 - Allowed with an exact match
-* 2 - Allowed with a prefix match
-
-For permissions options 1 & 2, we also need to specify the prefix as a byte string.
-
-* Prefix - byte string
-
-Specifying a permission type of 2 or 3 and then not supplying a byte string, or supplying a 0 length byte
-string is equivalent to Blocking that action.
-
-As an alternative to carrying two attributes values for each action, we could code these into a single number to
-save wire size.
-
-
-|---------------|-------------------------------------------------------------|
-|  Code Point   |                    Definition                               |
-|---------------|-------------------------------------------------------------|
-|     0x01      | ANNOUNCE - Allowed for all Namespaces/Names                 |
-|     0x02      | ANNOUNCE - Allowed with an exact match                      |
-|     0x03      | ANNOUNCE - Allowed with a prefix match                      |
-|     0x04      | SUBSCRIBE_ANNOUNCES - Allowed for all Namespaces/Names      |
-|     0x05      | SUBSCRIBE_ANNOUNCES - Allowed with an exact match           |
-|     0x06      | SUBSCRIBE_ANNOUNCES - Allowed with a prefix match           |
-|     0x07      | PUBLISH - Allowed for all Namespaces/Names                  |
-|     0x08      | PUBLISH - Allowed with an exact match                       |
-|     0x09      | PUBLISH - Allowed with a prefix match                       |
-|     0x0A      | FETCH - Allowed for all Namespaces/Names                    |
-|     0x0B      | FETCH - Allowed with an exact match                         |
-|     0x0C      | FETCH - Allowed with a prefix match                         |
-|---------------|-------------------------------------------------------------|
-
+The "moqt" claim limits the MOQT actions for which the token can provide
```

Comment:
just an observation, can we call it "scope" or "authz-scope" or "authz-cover" instead of limit.